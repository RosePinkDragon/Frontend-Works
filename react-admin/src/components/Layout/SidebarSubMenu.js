import React, { useState } from "react";

const SidebarSubMenu = ({ data }) => {
  const [isActive, setIsActive] = useState(false);

  return (
    <>
      {data && (
        <>
          <div className="subMenu">
            <p onClick={() => setIsActive(!isActive)}>{data.name}</p>
          </div>
          <div className="subData">
            {isActive
              ? data.options.map((option, idx) => (
                  <p key={idx} onClick={() => setIsActive(!isActive)}>
                    {option}
                  </p>
                ))
              : ""}
          </div>
        </>
      )}
    </>
  );
};

export default SidebarSubMenu;
