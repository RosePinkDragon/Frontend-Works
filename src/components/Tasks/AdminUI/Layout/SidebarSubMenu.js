import React, { useState } from "react";

const SidebarSubMenu = ({ data }) => {
  const [isActive, setIsActive] = useState(false);

  return (
    <>
      {data && (
        <>
          <div className="subMenu">
            <p onClick={() => setIsActive(!isActive)}>{data.name}</p>
          </div>
          {isActive ? (
            <div className="subData">
              {data.options.map((option, idx) => (
                <p key={idx} onClick={() => setIsActive(!isActive)}>
                  {option}
                </p>
              ))}
            </div>
          ) : (
            ""
          )}
        </>
      )}
    </>
  );
};

export default SidebarSubMenu;
