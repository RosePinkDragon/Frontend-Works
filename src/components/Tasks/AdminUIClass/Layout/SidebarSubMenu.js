import React from "react";

class SidebarSubMenu extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isActive: false,
    };
  }

  render() {
    return (
      <>
        {this.props.data && (
          <>
            <div className="subMenu">
              <p
                onClick={() =>
                  this.setState({ isActive: !this.state.isActive })
                }
              >
                {this.props.data.name}
              </p>
            </div>
<<<<<<< Updated upstream
            <div className="subData">
              {this.state.isActive
                ? this.data.options.map((option, idx) => (
                    <p
                      key={idx}
                      onClick={() =>
                        this.setState({ isActive: !this.state.isActive })
                      }
                    >
                      {option}
                    </p>
                  ))
                : ""}
            </div>
=======
            {this.state.isActive ? (
              <div className="subData">
                {this.props.data.options.map((option, idx) => (
                  <p
                    key={idx}
                    onClick={() =>
                      this.setState({ isActive: !this.state.isActive })
                    }
                  >
                    {option}
                  </p>
                ))}
              </div>
            ) : (
              ""
            )}
>>>>>>> Stashed changes
          </>
        )}
      </>
    );
  }
}

export default SidebarSubMenu;
