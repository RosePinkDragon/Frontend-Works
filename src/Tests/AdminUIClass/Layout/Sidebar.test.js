import { mount } from "enzyme";
import SidebarSubMenu from "components/Tasks/AdminUIClass/Layout/SidebarSubMenu";

// const title = "Test Title";
const data = {
  name: "Admin Menu",
  options: ["All Admins", "This Admin", "None"],
};

let wrapped = mount(<SidebarSubMenu data={data} />);
describe("SidebarSubMenu Render", () => {
  it("should render the Title Component correctly", () => {
    expect(wrapped).toMatchSnapshot();
  });
});

describe("updates state onClick", () => {
  let wrapper = mount(<SidebarSubMenu data={data} />);

  it("should not render subData menu", () => {
    expect(wrapper.find(".subData").length).toEqual(0);
  });

  it("should render subData menu", () => {
    wrapper.find("p").simulate("click");
    expect(wrapper.find(".subData").length).toEqual(1);
    expect(wrapper.find(".subData").find("p").length).toEqual(
      data.options.length
    );
  });

  it("should remove subData menu", () => {
    wrapper.find("p").at(1).simulate("click");
    expect(wrapper.find(".subData").length).toEqual(0);
    expect(wrapper.find(".subData").find("p").length).toEqual(0);
  });
});
