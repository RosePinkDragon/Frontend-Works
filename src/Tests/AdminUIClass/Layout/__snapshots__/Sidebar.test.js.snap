// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarSubMenu Render should render the Title Component correctly 1`] = `
<SidebarSubMenu
  data={
    Object {
      "name": "Admin Menu",
      "options": Array [
        "All Admins",
        "This Admin",
        "None",
      ],
    }
  }
>
  <div
    className="subMenu"
  >
    <p
      onClick={[Function]}
    >
      Admin Menu
    </p>
  </div>
</SidebarSubMenu>
`;
